### PR TITLE
Change proxy and proxied checks

### PR DESCRIPTION
**Pull request by jcortejoso**
### Description

Fixes for celostats integration

### Tested

Tested on a testnet

### CODE REVIEW

File: consensus/istanbul/backend/backend.go [hunk 1]
```diff
@@ -273,18 +273,20 @@ type Backend struct {
...
 func (sb *Backend) IsProxiedValidator() bool {
-	return sb.proxyNode != nil && sb.proxyNode.peer != nil
+	return sb.config.Proxied
 }
 
 // SendDelegateSignMsgToProxy sends an istanbulDelegateSign message to a proxy
 // if one exists
 func (sb *Backend) SendDelegateSignMsgToProxy(msg []byte) error {
-	if !sb.IsProxiedValidator() {
+	if !sb.IsProxiedValidator() || sb.proxyNode.peer == nil {
```
Comment (timmoreton) [inline consensus/istanbul/backend/backend.go:289]: panic if `sb.ProxyNode == nil` here ?

Comment (tkporter) [reply consensus/istanbul/backend/backend.go:289]: I could see this state happening if a validator & proxy disconnection happens right after a validator receives a delegate sign message from the proxy, so I don't think we should purposely cause a validator to crash

Comment (timmoreton) [reply consensus/istanbul/backend/backend.go:289]: what I mean is it would panic if `sb.ProxyNode == nil`

Comment (timmoreton) [reply consensus/istanbul/backend/backend.go:289]: consider ```if !sb.IsProxiedValidator() || sb.proxyNode == nil || sb.proxyNode.peer == nil {```

Comment (tkporter) [reply consensus/istanbul/backend/backend.go:289]: Ahh good catch

**Pull request merged**